sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -667,10 +715,12 @@ public void validate(PBegin input) {
 
       @Override
       public PCollection<TableRow> expand(PBegin input) {
-        String uuid = randomUUIDString();
-        final String jobIdToken = "beam_job_" + uuid;
-
+        stepUuid = randomUUIDString();
         BigQueryOptions bqOptions = input.getPipeline().getOptions().as(BigQueryOptions.class);
+        jobUuid = NestedValueProvider.of(
+           StaticValueProvider.of(bqOptions.getJobName()), new CreatePerBeamJobUuid(stepUuid));
```

Comment:
This will not actually solve the issue you're trying to solve, as we are still creating / accessing both the job name and step name statically in `expand`. This means that in a templatized pipeline, these values will still be constant.

Do you want to also VP-ify the job name? I guess that is problematic because that affects all runners?

cc: @tgroh